cmd/thanos/query.go

```diff
@@ -529,12 +573,72 @@ func runQuery(
 		})
 	}
 
-	grpcProbe := prober.NewGRPC()
-	httpProbe := prober.NewHTTP()
-	statusProber := prober.Combine(
-		httpProbe,
-		grpcProbe,
-		prober.NewInstrumentation(comp, logger, extprom.WrapRegistererWithPrefix("thanos_", reg)),
+	var (
+		allClients = func() []store.Client {
```

Comment:
It would be amazing to add a comment here to explain what we are doing and add a link to the proposal.